pkg/kptfile/kptfileutil/util.go

```diff
@@ -229,6 +241,70 @@ func UpdateUpstreamLockFromGit(path string, spec *git.RepoSpec) error {
 	return WriteFile(path, kpgfile)
 }
 
+func merge(localKf, updatedKf, originalKf kptfilev1alpha2.KptFile) (kptfilev1alpha2.KptFile, error) {
+	localBytes, err := yaml.Marshal(localKf)
+	if err != nil {
+		return kptfilev1alpha2.KptFile{}, err
+	}
+
+	updatedBytes, err := yaml.Marshal(updatedKf)
+	if err != nil {
+		return kptfilev1alpha2.KptFile{}, err
+	}
+
+	originalBytes, err := yaml.Marshal(originalKf)
+	if err != nil {
+		return kptfilev1alpha2.KptFile{}, err
+	}
+
+	mergedBytes, err := merge3.MergeStrings(string(localBytes), string(originalBytes), string(updatedBytes), false)
+	if err != nil {
+		return kptfilev1alpha2.KptFile{}, err
+	}
+
+	var mergedKf kptfilev1alpha2.KptFile
+	err = yaml.Unmarshal([]byte(mergedBytes), &mergedKf)
+	if err != nil {
+		return mergedKf, err
+	}
+
+	// The merge algorithm currently lets values from upstream take precedence,
+	// and we don't want that for name and namespace. So updating those values
+	// in the merge Kptfile.
+	mergedKf.ObjectMeta.Name = localKf.Name
+	mergedKf.ObjectMeta.Namespace = localKf.Namespace
```

Comment:
I am assuming namespace doesn't have any significance when it comes to pkg workflow.